libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt

```diff
@@ -20,41 +15,45 @@ class MapboxOffboardRouter(
 ) : Router {
 
     private var navigationRoute: NavigationRoute? = null
+    // For testing only
+    private var isNeedToBuildNavigationRoute = true
+    // For testing only
+    private lateinit var routeRetrieveCallback: RouteRetrieveCallback
 
     override fun getRoute(
         origin: Point,
         waypoints: List<Point>?,
         destination: Point,
         callback: Router.Callback
     ) {
-        val builder = NavigationRoute
-            .builder(context)
-            .accessToken(mapboxToken)
-            .origin(origin)
-            .destination(destination)
-        waypoints?.forEach { builder.addWaypoint(it) }
-        navigationRoute = builder.build()
-        navigationRoute?.getRoute(object : Callback<DirectionsResponse> {
-            override fun onFailure(call: Call<DirectionsResponse>, t: Throwable) {
-                callback.onFailure(t)
-            }
-
-            override fun onResponse(
-                call: Call<DirectionsResponse>,
-                response: Response<DirectionsResponse>
-            ) {
-                val route = response.body()?.routes()
-                if (response.isSuccessful && !route.isNullOrEmpty()) {
-                    callback.onRouteReady(route.map { it.mapToRoute() })
-                } else {
-                    callback.onFailure(NavigationException("Error fetching route"))
-                }
-            }
-        })
+        if (isNeedToBuildNavigationRoute) {
+            val builder = NavigationRoute
+                .builder(context)
+                .accessToken(mapboxToken)
+                .origin(origin)
+                .destination(destination)
+            waypoints?.forEach { builder.addWaypoint(it) }
+            navigationRoute = builder.build()
+        }
+        if (!::routeRetrieveCallback.isInitialized) {
+            routeRetrieveCallback = RouteRetrieveCallback(callback)
+        }
+        navigationRoute?.getRoute(routeRetrieveCallback)
     }
 
     override fun cancel() {
         navigationRoute?.cancelCall()
         navigationRoute = null
     }
+
+    // For testing only
+    internal fun setNavigationRoute(navigationRoute: NavigationRoute) {
+        this.navigationRoute = navigationRoute
+        isNeedToBuildNavigationRoute = false
+    }
+
+    // For testing only
+    internal fun setRouteRetrieveCallback(routeRetrieveCallback: RouteRetrieveCallback) {
```

Comment:
This is not necessary. You can capture the callback instead and invoke it in the test block itself. Take a look at:
https://github.com/mapbox/mapbox-navigation-android/blob/master/libdirections-session/src/test/java/com/mapbox/navigation/directions/session/MapboxDirectionsSessionTest.kt#L30-L32

Above gives you the reference to the callback whenever you need it to verify behavior. In order to achieve it, we first need to mock the `NavigationRoute`.